pkg/deploy/controller/deployment/controller.go

```diff
@@ -76,12 +85,15 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		podName := deployment.Annotations[deployapi.DeploymentPodAnnotation]
 		if err := c.podClient.deletePod(deployment.Namespace, podName); err != nil {
 			if !kerrors.IsNotFound(err) {
+				// TODO: create event?
```

Comment:
Related: at what point should the deployment enter a failed state? Currently the only way it can enter a terminal state is by the completion of the deployer pod, but if the deployer pod never exists and we are not going to retry, seems like maybe we should set a terminal state.